src/v/cluster/partition_balancer_planner.cc

```diff
@@ -395,26 +395,49 @@ void partition_balancer_planner::get_full_node_reassignments(
     }
 }
 
-std::vector<ntp_reassignments>
-partition_balancer_planner::get_ntp_reassignments(
+void partition_balancer_planner::get_unavailable_node_movement_cancelations(
+  std::vector<model::ntp>& cancelations,
+  const reallocation_request_state& rrs) {
+    for (auto& update : _topic_table.in_progress_updates()) {
+        if (
+          update.second.state
+          != topic_table::in_progress_state::update_requested) {
+            continue;
+        }
+        auto current_assignments = _topic_table.get_partition_assignment(
+          update.first);
+        if (!current_assignments.has_value()) {
+            continue;
+        }
+        for (auto& r : current_assignments->replicas) {
```

Comment:
we should check if node was not present in previous assignments - if it was, no need to cancel